src/gui/attributetable/qgsfeaturelistmodel.cpp

```diff
@@ -293,11 +295,18 @@ bool QgsFeatureListModel::sortByDisplayExpression() const
 
 void QgsFeatureListModel::setSortByDisplayExpression( bool sortByDisplayExpression, Qt::SortOrder order )
 {
+  if ( !mSortByDisplayExpression && sortByDisplayExpression )
+  {
+    masterModel()->prefetchSortData( mDisplayExpression.expression(), 1 );
```

Comment:
1 the cache index, also maybe add a variable used across this file here?

Reply:
@3nids , done.